brainscore/metrics/accuracy_delta.py

```diff
@@ -0,0 +1,108 @@
+import itertools
+
+import numpy as np
+
+from brainscore.metrics import Metric, Score
+import random
+from brainio.assemblies import BehavioralAssembly
+
+
+# controls how many half-splits are averaged together to get human delta.
+HUMAN_SPLITS = 100
+
+
+class AccuracyDelta(Metric):
+
+    def __init__(self, image_types: list):
+        self.image_types = image_types
+
+    def __call__(self, source: BehavioralAssembly, target: BehavioralAssembly):
+
+        model_delta = get_model_delta(source, self.image_types)
+        scores = []
+
+        # calculate score over average of 100 sub splits of human delta
+        for i in range(HUMAN_SPLITS):
+            human_delta = get_human_delta(target, self.image_types, random_state=i)
+            score = max((1 - ((np.abs(human_delta - model_delta)) / human_delta)), 0)
+            scores.append(score)
+        score = np.mean(scores)
+        error = np.std(scores)
+
+        return Score([score, error], coords={'aggregation': ['center', 'error']}, dims=('aggregation',))
+
+
+def extract_subjects(assembly):
+    return list(sorted(set(assembly['subject'].values)))
+
+
+def get_human_delta(target, image_types, random_state=0, isCeiling=False):
+    # calculate human accuracies for [whole, condition]
+    condition_scores_human = {}
+
+    if isCeiling:
+        half1 = target
+    else:
+        # pull half of subjects. This is outside loop to make sure
+        # the same half of subjects are used in w and f conditions.
+        random_state = np.random.RandomState(random_state)
+        num_subjects = len(set(target["subject"].values))
+        half1_subjects = random_state.choice(range(1, num_subjects), (num_subjects // 2), replace=False)
+        half1 = target[{'presentation': [subject in half1_subjects for subject in target['subject'].values]}]
+
+    # for whole condition, and other condition (frank or frag)
+    for image_type in image_types:
+        scores = []
+
+        # get per subject accuracy
+        for subject in extract_subjects(half1):
+            this_target = target.sel(image_type=image_type, subject=subject)
+            correct_count = np.count_nonzero(this_target.values)
+            accuracy = correct_count / len(this_target)
+            scores.append(accuracy)
+        condition_scores_human[image_type] = scores
+
+    # calculate 16-human pairwise accuracy:
+    condition = image_types[1]
+    delta_vector = [a_i - b_i for a_i, b_i in zip(condition_scores_human["w"], condition_scores_human[condition])]
```

Comment:
Would this be simpler?

```suggestion
    subject_deltas = []
    for subject in half1_subjects:
        condition1_responses = target.sel(**{self.condition_key: self.condition_values[0], 'subject': subject)
        condition1_accuracy = np.mean(condition1_responses)  # response values are either 1 for correct or 0 for incorrect
        condition2_responses = target.sel(**{self.condition_key: self.condition_values[1], 'subject': subject)
        condition2_accuracy = np.mean(condition2_responses)
        subject_delta = condition1_accuracy - condition2_accuracy
        subject_deltas.append(subject_delta)
```